app/controllers/subscribers_from_account_token_controller.rb

```diff
@@ -0,0 +1,79 @@
+class SubscribersFromAccountTokenController < ApplicationController
+  before_action :validate_params
+
+  class AccountsDown < RuntimeError; end
+  class TokenInvalid < RuntimeError; end
+
+  rescue_from AccountsDown do |exception|
+    render json: { error: "Could not fetch subscriber",
+                   details: exception.message },
+           status: :internal_server_error
+  end
+
+  rescue_from TokenInvalid do |exception|
+    render json: { error: "Could not fetch subscriber",
+                   details: exception.message },
+           status: :forbidden
+  end
+
+  def from_token
+    address = expected_params.require(:address)
+    token = expected_params.require(:token)
+
+    validate_token!(address, token)
+
+    subscriber = find_subscriber(address)
+
+    render json: { subscriber: subscriber }
+  end
+
+private
+
+  def authorise
+    authorise_user!("manage_subscriptions_from_account")
+  end
+
+  def validate_token!(address, token)
+    account_manager = Plek.find("account-manager")
+    userinfo_endpoint = rest_call("#{account_manager}/.well-known/openid-configuration")["userinfo_endpoint"]
+    userinfo = rest_call(userinfo_endpoint, token)
+
+    raise TokenInvalid, "email address does not match" unless userinfo["email"] == address
+    raise TokenInvalid, "email address not verified" unless userinfo["email_verified"]
+  end
+
+  def rest_call(uri, token = nil)
+    headers = { accept: :json }
+    headers.merge!(authorization: "Bearer #{token}") if token
+    response = RestClient.get(uri, headers)
+    JSON.parse(response.body)
+  rescue RestClient::RequestFailed, JSON::ParserError
+    raise AccountsDown, "failed rest call"
+  end
+
+  def find_subscriber(address)
+    Subscriber.resilient_find_or_create(
+      address,
+      signon_user_uid: current_user.uid,
+    ).tap do |subscriber|
+      subscriber.activate if subscriber.deactivated?
+    end
+  end
+
+  def expected_params
+    params.permit(:address, :token)
+  end
+
+  def validate_params
+    ParamsValidator.new(expected_params).validate!
+  end
+
+  class ParamsValidator < OpenStruct
+    include ActiveModel::Validations
+
+    validates :address, presence: true
+    validates :address, email_address: true
```

Comment:
Minor, this can just be `validates :address, presence: true, email_address: true`